src/pyfmi/fmi.pyx

```diff
@@ -6969,10 +6992,10 @@ cdef class FMUModelCS2(FMUModelBase2):
 
         if self._fmu_kind != FMIL.fmi2_fmu_kind_cs:
             if self._fmu_kind != FMIL.fmi2_fmu_kind_me_and_cs:
-                raise FMUException("This class only supports FMI 2.0 for Co-simulation.")
+                raise InvalidVersionException("This class only supports FMI 2.0 for Co-simulation.")
 
         if self.get_capability_flags()['needsExecutionTool'] == True:
-            raise FMUException('Models that need an execution tool are not supported')
+            raise InvalidVersionException('Models that need an execution tool are not supported')
```

Comment:
Why should this be an InvalidVersionException? needsExecutionTool has nothing to do with version.